mysql/toddl.go

```diff
@@ -0,0 +1,186 @@
+		spTableName, err := internal.GetSpannerTable(conv, srcTable.Name)
+		if err != nil {
+			conv.Unexpected(fmt.Sprintf("Couldn't map source table %s to Spanner: %s", srcTable.Name, err))
+			continue
+		}
+		var spColNames []string
+		spColDef := make(map[string]ddl.ColumnDef)
+		conv.Issues[srcTable.Name] = make(map[string][]internal.SchemaIssue)
+		// Iterate over columns using ColNames order.
+		for _, srcColName := range srcTable.ColNames {
+			srcCol := srcTable.ColDefs[srcColName]
+			colName, err := internal.GetSpannerCol(conv, srcTable.Name, srcCol.Name, false)
+			if err != nil {
+				conv.Unexpected(fmt.Sprintf("Couldn't map source column %s of table %s to Spanner: %s", srcTable.Name, srcCol.Name, err))
+				continue
+			}
+			spColNames = append(spColNames, colName)
+			ty, issues := toSpannerType(conv, srcCol.Type.Name, srcCol.Type.Mods)
+			if len(srcCol.Type.ArrayBounds) > 1 {
+				ty = ddl.String{Len: ddl.MaxLength{}}
+				issues = append(issues, internal.MultiDimensionalArray)
+			}
+			// TODO(hengfeng): add issues for all elements of srcCol.Ignored.
+			if srcCol.Ignored.ForeignKey {
+				issues = append(issues, internal.ForeignKey)
+			}
+			if srcCol.Ignored.Default {
+				issues = append(issues, internal.DefaultValue)
+			}
+			if len(issues) > 0 {
+				conv.Issues[srcTable.Name][srcCol.Name] = issues
+			}
+			spColDef[colName] = ddl.ColumnDef{
+				Name:    colName,
+				T:       ty,
+				IsArray: len(srcCol.Type.ArrayBounds) == 1,
+				NotNull: srcCol.NotNull,
+				Comment: "From: " + quoteIfNeeded(srcCol.Name) + " " + srcCol.Type.Print(),
+			}
+		}
+		comment := "Spanner schema for source table " + quoteIfNeeded(srcTable.Name)
+		conv.SpSchema[spTableName] = ddl.CreateTable{
+			Name:     spTableName,
+			ColNames: spColNames,
+			ColDefs:  spColDef,
+			Pks:      cvtPrimaryKeys(conv, srcTable.Name, srcTable.PrimaryKeys),
+			Comment:  comment}
+	}
+	return nil
+}
+
+// toSpannerType maps a scalar source schema type (defined by id and
+// mods) into a Spanner type. This is the core source-to-Spanner type
+// mapping.  toSpannerType returns the Spanner type and a list of type
+// conversion issues encountered.
+func toSpannerType(conv *internal.Conv, id string, mods []int64) (ddl.ScalarType, []internal.SchemaIssue) {
+	maxExpectedMods := func(n int) {
+		if len(mods) > n {
+			conv.Unexpected(fmt.Sprintf("Found %d mods while processing type id=%s", len(mods), id))
+		}
+	}
+	switch id {
+	case "bool", "boolean":
+		maxExpectedMods(0)
+		return ddl.Bool{}, nil
+	case "tinyint":
+		maxExpectedMods(1)
+		// tinyint(1) is a bool in MySQL
+		if len(mods) > 0 && mods[0] == 1 {
+			return ddl.Bool{}, nil
+		}
+		return ddl.Int64{}, []internal.SchemaIssue{internal.Widened}
+	case "double":
+		maxExpectedMods(2)
+		return ddl.Float64{}, nil
+	case "float":
+		maxExpectedMods(2)
+		return ddl.Float64{}, []internal.SchemaIssue{internal.Widened}
+	case "numeric", "decimal": // Map all numeric and decimal types to float64.
```

Comment:
Correct, both are same in MySQL. But NUMERIC is generally not used in MySQL, it's used as DECIMAL only. Also, if NUMERIC is used by any chance, then pingcap parser converts it to DECIMAL while parsing. 
I have just used DecimalThatFits and Decimal instead of NumericThatFits and Numeric to provide description according to Numeric/Decimal type.
https://github.com/cloudspannerecosystem/harbourbridge/blob/99db369b2b826fc24ffbb760f3b3700cd0e04b0f/internal/report.go#L249-L250
It's not much different, just replacing `numeric` keyword with `decimal`. If you think that it would be better to just use NumericThatFits and Numeric and also change description as:
```
Numeric:               {brief: "Spanner does not support numeric/decimal. This type mapping could lose precision and is not recommended for production use", severity: warning},
	NumericThatFits:       {brief: "Spanner does not support numeric/decimal, but this type mapping preserves the numeric/decimal's specified precision", severity: note},
```

What are your views on this?

Reply:
That makes sense.
Let's leave as is.